### ISSUE TITLE
validate_params insists that all parameters be of the same length

### ISSUE THREAD

**Issue by slyrus** [r-dbi/RPostgres#305]
I'm trying to pass two arguments to a query. One is a list to be used in the form of SELECT FOO IN ($1) while the second parameter is used just for something like BAR = $2. 

Why should these two parameters need to be of the same length? A workaround is to replicate the other arguments, but this seems wrong.

**Comment by krlmlr** [comment]
DBI supports passing multiple parameter sets, e.g. `params = list(1:2, 3:4)` runs the query with parameters (1, 3) and (2, 4). Can you use `list(SQL("'foo1', 'foo2'"), "bar")` ?

**Comment by slyrus** [comment]
Right, I get that there can be multiple parameter sets, but why should they be of the same length? If I'm doing "SELECT * FROM a, b WHERE a.ID in ($!) and b.ID in ($2)", I see no reason why $1 and $2 should be of the same length. I guess I don't get what you're implying happens with, e.g., parameters (1,3) and (2,4). I would expect that this query would return rows corresponding to (1,2), (1,4), (3, 2), and (3,4) not just (1,2) and (3,4). Am I missing something?

**Comment by krlmlr** [comment]
There is currently no way to pass a list of values to be used in an `IN` statement. Can you use `dbQuoteLiteral()` and `paste(collapse = ", ")` to construct the query?

Regarding multiple parameter sets:

``` r
library(RPostgres)

con <- dbConnect(Postgres())

dbGetQuery(con, "SELECT $1 AS a, $2 AS b", list(1:2, 3:4))
#>   a b
#> 1 1 3
#> 2 2 4
```

<sup>Created on 2021-08-24 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

**Comment by slyrus** [comment]
I suppose I _could_, but I'd rather consider this issue to be a feature request to be able to pass lists in directly. And (forgive me as it's been a couple months), IIRC, the problem isn't that i can't pass in lists, but rather that there is a specious check that the passed in lists be of the same length.
